genestack_client/settings/genestack_config.py

```diff
@@ -194,5 +170,63 @@ def save(self):
         with open(config_path, 'w') as f:
             document.writexml(f, indent='', addindent='    ', newl='\n')
 
+    def _store_value_insecurely(self, value, document, user_element, element_name):
+        """
+        Store value in xml config.
+
+        :param value: value to be stored
+        :type value: basestring
+        :param document: document
+        :type document: xml.dom.minidom.Document
+        :param user_element: parent element
+        :type user_element: xml.dom.minidom.Element
+        :param element_name: name of the element
+        :type element_name: basestring
+        :return: None
+        """
+        if self.store_raw is not None:
+            save_to_file = self.store_raw
+        elif self.store_raw_session is not None:
+            save_to_file = self.store_raw_session
+        else:
+            try:
+                save_to_file = ask_confirmation(
+                    'Do you want to store secure value in config file as plain text?',
+                    default='n')
+            except KeyboardInterrupt:
+                save_to_file = False
+            try:
+                self.store_raw = ask_confirmation('Set this as default behaviour?', default='y')
+            except KeyboardInterrupt:
+                self.store_raw_session = save_to_file
+        if save_to_file:
+            value_element = document.createElement(element_name)
+            value_element.appendChild(document.createTextNode(value))
+            user_element.appendChild(value_element)
+        else:
+            sys.stderr.write('"%s" was not saved to file\n' % element_name)
+
+    def _store_value_securely(self, alias, secret_value, key):
+        """
+        Try to store value in security vault.
+
+        :param alias: user alias
+        :type alias: basestring
+        :param secret_value: value to be stored
+        :type secret_value: basestring
+        :param key: key for storage
+        :type key: basestring
+        :return: ``True`` if stored successfully
+        :rtype: bool
+        """
+        try:
+            import keyring
+            keyring.set_password(key, alias, secret_value)
+            return True
+        except Exception as e:
+            if not self.store_raw:
+                sys.stderr.write('Cannot store in secure storage: %s\n' % e)
+            return False
```

Comment:
I've just thought: wouldn't raising error from within this function be more logical and Pythonic than returning `bool`?  So instead of
```py
if not self._store_value_securely(user.alias, user.password, GENESTACK_SDK):
    self._store_value_insecurely(user.password, document, user_element, 'password')
```
we could write
```py
try:
    self._store_value_securely(user.alias, user.password, GENESTACK_SDK)
except GenestackException:
    self._store_value_insecurely(user.password, document, user_element, 'password')
```
What do you think?